### PR TITLE
Clean up L1TTrackMatch producers

### PR DESCRIPTION
**Pull request by kpedro88**
#### PR description:

@Dr15Jones pointed out that the producers introduced in #30363 were not converted from legacy producers to modern producers. I converted them and applied the usual code review comments that were skipped/ignored in the original PR:
* delete commented-out code
* use std::make_unique, std::abs, reco::deltaR, reco::reduceRange
* member variables end in underscore
* range-based loops
* capture returned temporaries with const auto&
* etc.

#### PR validation:

Ran workflow 23234.0 successfully.

Should this be backported to 11_1_X?

### CODE REVIEW

File: L1Trigger/L1TTrackMatch/plugins/L1TkFastVertexProducer.cc [hunk 1]
```diff
@@ -148,28 +137,28 @@ L1TkFastVertexProducer::~L1TkFastVertexProducer() {
...
-void L1TkFastVertexProducer::produce(edm::Event& iEvent, const edm::EventSetup& iSetup) {
+void L1TkFastVertexProducer::produce(edm::StreamID, edm::Event& iEvent, const edm::EventSetup& iSetup) const {
   using namespace edm;
 
-  std::unique_ptr<TkPrimaryVertexCollection> result(new TkPrimaryVertexCollection);
+  auto result = std::make_unique<TkPrimaryVertexCollection>();
 
   // Tracker Topology
   edm::ESHandle<TrackerTopology> tTopoHandle_;
   iSetup.get<TrackerTopologyRcd>().get(tTopoHandle_);
```
Comment (Dr15Jones) [inline L1Trigger/L1TTrackMatch/plugins/L1TkFastVertexProducer.cc:147]: Could we also get esConsumes used with the appropriate token?

Comment (kpedro88) [reply L1Trigger/L1TTrackMatch/plugins/L1TkFastVertexProducer.cc:147]: Sure, I'll do that now.

**Comment by kpedro88** [comment]
please test

**Comment by kpedro88** [comment]
+upgrade

**Comment by kpedro88** [comment]
@rekovic please take a look

**Comment by silviodonato** [comment]
merge

**Pull request merged**